include/CL/sycl/handler.hpp

```diff
@@ -66,8 +66,19 @@ template<int dimensions, class Function>
 __global__ void parallel_for_kernel(Function f,
                                     sycl::range<dimensions> execution_range)
 {
-  item<dimensions, false> this_item;
-  if(this_item.get_linear_id() < execution_range.size())
+  item<dimensions, false> this_item{item_impl<dimensions>{execution_range}};
+
+  bool item_is_in_range = true;
+  for(int i = 0; i < dimensions; ++i)
+  {
+    if(this_item.get_id(i) >= execution_range.get(i))
+    {
+      item_is_in_range = false;
+      break;
+    }
+  }
```

Comment:
Could we perhaps add a function so that we can share code between `parallel_for_kernel` and the similar implementation in `parallel_for_kernel_with_offset`?

Reply:
Sure, I'll look into it!

Reply:
Thanks!